src/main/java/net/javadiscord/javabot/systems/moderation/ReportCommand.java

```diff
@@ -68,23 +157,26 @@ private EmbedBuilder buildReportEmbed(User reported, User reportedBy, TextChanne
 				.addField("Channel", channel.getAsMention(), true)
 				.addField("Reported on", String.format("<t:%s:F>", Instant.now().getEpochSecond()), false)
 				.addField("ID", String.format("```%s```", reported.getId()), true)
+				.addField("Reason", String.format("```%s```", reason), false)
 				.setFooter(reportedBy.getAsTag(), reportedBy.getEffectiveAvatarUrl())
 				.setTimestamp(Instant.now());
 	}
 
 	@Override
 	protected ReplyCallbackAction handleModerationActionCommand(SlashCommandInteractionEvent event, Member commandUser, Member target) throws ResponseException {
-		OptionMapping option = event.getOption("reason");
+		OptionMapping option = event.getOption(REASON_OPTION_NAME);
 		String reason = option == null ? "None" : option.getAsString();
-
+		Member member = event.getOption("user").getAsMember();
+		if(member == null) {
+			return Responses.error(event, "Cannot report a user who is not a member of this server");
+		}
 		var config = Bot.config.get(event.getGuild());
 		MessageChannel reportChannel = config.getModeration().getReportChannel();
-		var embed = buildReportEmbed(target.getUser(), commandUser.getUser(), event.getTextChannel(), config.getSlashCommand());
-		embed.addField("Reason", String.format("```%s```", reason), false);
+		var embed = buildReportEmbed(member.getUser(), reason, event.getUser(), event.getTextChannel(), config.getSlashCommand());
 		reportChannel.sendMessage("@here").setEmbeds(embed.build())
-				.setActionRows(setComponents(target.getIdLong()))
+				.setActionRows(setComponents(member.getIdLong()))
 				.queue();
-		embed.setDescription("Successfully reported " + "`" + target.getUser().getAsTag() + "`!\nYour report has been send to our Moderators");
+		embed.setDescription("Successfully reported " + "`" + member.getUser().getAsTag() + "`!\nYour report has been send to our Moderators");
 		return event.replyEmbeds(embed.build()).setEphemeral(true);
```

Comment:
Couldn't you just use the `target` member?

Reply:
I have now (c617c18a992d4eb58b5737c9ed250eaba0fb8cb5) changed this and applied the stuff introduced by #222